container/libcontainer/compatibility.go

```diff
@@ -158,9 +192,17 @@ func ReadConfig(dockerRoot, dockerRun, containerID string) (*configs.Config, err
 		}
 
 		var state libcontainer.State
-		err = json.Unmarshal(out, &state)
-		if err != nil {
-			return nil, err
+		if err = json.Unmarshal(out, &state); err != nil {
+			if _, ok := err.(*json.UnmarshalTypeError); ok {
```

Comment:
Yeah, I checked most of codes which used `json.UnmarshalTypeError`, it should be pointer type.